R/qualification-ddi.R

```diff
@@ -217,11 +218,43 @@ generateDDIQualificationDDIPlot <- function(data) {
   ddiPlotConfiguration$export$height <- 2 * 1.2 * (data$plotSettings$height / 96)
   ddiPlotConfiguration$export$units <- "in"
 
+  # Set axis label font size
   ddiPlotConfiguration$labels$xlabel$font$size <- 2 * data$plotSettings$axisFontSize
   ddiPlotConfiguration$labels$ylabel$font$size <- 2 * data$plotSettings$axisFontSize
+
+  # Set axis tick font size
+  ddiPlotConfiguration$xAxis$font$size <- 2 * data$plotSettings$axisFontSize
+  ddiPlotConfiguration$yAxis$font$size <- 2 * data$plotSettings$axisFontSize
+
+  # Set watermark font size
   ddiPlotConfiguration$background$watermark$font$size <- 2 * data$plotSettings$watermarkFontSize
+
+  # Set legend font size
   ddiPlotConfiguration$legend$font$size <- 2 * data$plotSettings$legendFontSize
 
+  # Set line color and type
+  ddiPlotConfiguration$lines$color <- "black"
+  ddiPlotConfiguration$lines$linetype <- "solid"
+
+  # Set axes scaling
+  if (data$axesSettings$X$scaling == "Log") {
+    ddiPlotConfiguration$xAxis$scale <- tlf::Scaling$log
+  }
+  if (data$axesSettings$Y$scaling == "Log") {
+    ddiPlotConfiguration$yAxis$scale <- tlf::Scaling$log
+  }
+
+  # Set y axis ticks and limits
+  if (residualsVsObservedFlag[[data$axesSettings$plotType]] & data$axesSettings$Y$scaling == "Log" ) {
+    log10Limit <- max(abs(sapply(c(floor, ceiling), function(fn) fn(log10(ddiData[[data$axesSettings$Y$label]])))))

```

Comment:
What does this line do?

Reply:
Calculation of the maximal y axis limit in log scale.  This has now been split into three lines:
1) Find lower extreme of log10(predict/observed), find its lower bounding y axis tick (in log10 scale)
2) Find upper extreme of log10(predict/observed), find its upper bounding y axis tick  (in log10 scale)
3) Maximal y axis limit is the maximum of the absolute values from 1 and 2.